packages/xod-arduino/src/transpiler.js

```diff
@@ -121,11 +127,12 @@ const formatValueLiteral = def(
 
 // Creates a TConfig object from entry-point path and project
 const createTConfig = def(
-  'createTConfig :: PatchPath -> Project -> TConfig',
-  (path, project) => R.applySpec({
+  'createTConfig :: TranspilationOptions -> PatchPath -> Project -> TConfig',
+  (opts, path, project) => R.applySpec({
     NODE_COUNT: R.compose(getNodeCount, Project.getPatchByPathUnsafe(path)),
     MAX_OUTPUT_COUNT: getOutputCount,
-    XOD_DEBUG: R.F,
+    XOD_DEBUG: () => (opts.debug),
+    XOD_DEBUG_RUNTIME: () => (opts.debugRuntime),
```

Comment:
To KISS it would be much better to have only `debug` field in the options. `debugRuntime` is quite pointless. It shouldn’t be mentioned in JS. We can keep `//#define XOD_DEBUG_ENABLE_TRACE` commented out right in the template, unconditionally.